portlets/src/main/webapp/vue-app/engagement-center/components/EngagementCenter.vue

```diff
@@ -98,11 +98,20 @@ export default {
     });
     this.$root.$on('close-program-detail', () => this.displayProgramDetail = false);
     const urlPath = document.location.search || document.location.pathname;
-    const challengeId = urlPath.match( /\d+/ ) && urlPath.match( /\d+/ ).join('');
+    const id = urlPath.match( /\d+/ ) && urlPath.match( /\d+/ ).join('');
     if (urlPath.indexOf(`${eXo.env.portal.context}/${eXo.env.portal.portalName}/contributions/programs`) > -1) {
+      if (id) {
+        this.$programsServices.getProgramById(id)
+          .then(program => {
+            if (program && program.id) {
+              this.$root.$emit('open-program-detail', program);
+              window.history.replaceState('programs', this.$t('engagementCenter.label.programs'), `${eXo.env.portal.context}/${eXo.env.portal.portalName}/contributions/programs/${program.id}`);
```

Comment:
should be added in the created of [ProgramDetail.vue](https://github.com/Meeds-io/gamification/pull/513/files#diff-334e482cb6654f828c52b3a41a9144787a773582603187eeef30a8c89f60ee48)